Rebasing to native cluster and track

shahor02 opened the pull request.

shahor02 committed:
Rebase reconstruction to native cluster and track classes + related changes

shahor02 commented:
@fprino @galocco 
in principle, the reconstruction works with this PR, producing the same results as the main branch on 30 pp pythia events of https://cernbox.cern.ch/files/spaces/eos/user/s/shahoian/tmp/savPPP30.tar.gz (VT reco only, for the MS both main and this branches see nothing with so few pp events). I am not merging it yet because in some cases the cov.matrix gets wrong correlations, leading to negative chi2, will need some extra debugging. 
If you are planning some developments in the few following days, better to do them on this branch, to avoid the divergence: not sure I will manage to finish polishing tomorrow, then I'll travel til Jan.2.

fprino commented:
Hi Ruben.
Many thanks!
It's great progress!
I do not plan to work on new developments before January 3. I will mainly focus on testing the vertexer on the pp pythia events that you provided. If these tests will require some updates in the code, I'll do them on your development branch.

shahor02 committed:
TMP: to solve covmat instabilities

Debug mode added, to be removed.
Still need to eliminate cov.matrix non-pos def. at propagation and update

shahor02 commented:
rebased to account for https://github.com/shahor02/NA6PRoot/pull/45

shahor02 committed:
Add outward track param to NA6PTrack

shahor02 committed:
add output tree to runVTTrackFinderCA.C

At the moment the cov. matrix calculation is wrong, even with ad hoc stabilization

shahor02 commented:
Closing in favour of new PR
